src/routes/view-content/[guideId]/[bibleSection]/bible-menu/BookChapterSelectorPane.svelte

```diff
@@ -1,54 +1,39 @@
     import { Icon } from 'svelte-awesome';
     import arrowRight from 'svelte-awesome/icons/arrowRight';
     import type { ApiBibleBook, FrontEndVerseForSelectionPane, ApiBibleChapter } from '$lib/types/bible';
-    import { currentLanguageInfo } from '$lib/stores/language.store';
-    import type { Language } from '$lib/types/file-manager';
-    import { isOnline } from '$lib/stores/is-online.store';
-    import { preferredBibleIds } from '$lib/stores/preferred-bibles.store';
-    import { bibleChaptersByBookAvailableForGuide } from '$lib/utils/data-handlers/resources/guides';
-    import type { ApiParentResource } from '$lib/types/resource';
-    import { ContentTabEnum, getContentContext } from '../context';
-
-    const { currentGuide, currentTab, setCurrentTab, setCurrentBibleSection } = getContentContext();
-
-    export let bookChapterSelectorPane: CupertinoPane;
-    export let isShowing: boolean;
+    import { getContentContext } from '../context';
+    import { BookChapterSelectorPaneInfo, type ContentPaneInfo } from './pane-handler';
+
+    const { setCurrentBibleSectionAndCurrentGuide, currentPane, closeContextualMenu, closeCurrentPane } =
+        getContentContext();
+
     export let bookCodesToNames: Map<string, string> | undefined;
 
+    let pane: CupertinoPane | undefined;
+
     let currentBook: ApiBibleBook;
     let currentChapter: ApiBibleChapter | null = null;
     let buttons: HTMLButtonElement[] = [];
     let scrollBehaviorSmooth = false;
     let firstSelectedVerse: FrontEndVerseForSelectionPane | null = null;
     let lastSelectedVerse: FrontEndVerseForSelectionPane | null = null;
 
-    $: filterByCurrentGuide = $currentTab === ContentTabEnum.Guide;
-    $: availableBooksPromise = fetchAvailableBooks(
-        $currentLanguageInfo,
-        $currentGuide,
-        $isOnline,
-        filterByCurrentGuide,
-        $preferredBibleIds
-    );
+    let availableBooks: ApiBibleBook[] = [];
 
-    async function fetchAvailableBooks(
-        _currentLanguageInfo: Language | undefined,
-        currentGuide: ApiParentResource | null,
-        online: boolean,
-        filterByCurrentGuide: boolean,
-        preferredBibleIds: number[]
-    ) {
-        const allAvailable = await bibleChaptersByBookAvailable(online, preferredBibleIds);
-        if (filterByCurrentGuide && currentGuide?.id) {
-            return await bibleChaptersByBookAvailableForGuide(allAvailable, currentGuide.id);
-        } else {
-            return allAvailable;
+    $: showOrDismissPane($currentPane);
+
+    function showOrDismissPane(currentPane: ContentPaneInfo | null) {
+        if (currentPane instanceof BookChapterSelectorPaneInfo) {
+            availableBooks = currentPane.availableBooks;
+            pane?.present({ animate: true });
+        } else if (pane && pane.isPanePresented() && !pane.isHidden()) {
+            availableBooks = [];
+            pane.hide();
         }
     }
```

Comment:
All the calculating here that was done reactively based on various props is now done just-in-time when the user clicks to open the pane. This is more optimized and follows the pattern for the predetermined-passage selectors.